csFastFloat/FastFloatParser.cs

```diff
@@ -1,32 +1,37 @@
-﻿using csFastFloat.Enums;
-using csFastFloat.Structures;
-using System;
+﻿using System;
+using System.Diagnostics;
 using System.Numerics;
 using System.Runtime.CompilerServices;
-
-[assembly: InternalsVisibleTo("TestcsFastFloat")]
-
-[assembly: InternalsVisibleTo("Benchmark")]
-
+using System.Runtime.InteropServices;
+using csFastFloat.Enums;
+using csFastFloat.Structures;
 
 namespace csFastFloat
 {
   public static class FastFloatParser
   {
     private static void ThrowArgumentException() => throw new ArgumentException();
-    public static float exact_power_of_ten(long power) => Constants.powers_of_ten_float[power];
 
-    public static float ToFloat(bool negative, AdjustedMantissa am)
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static float exact_power_of_ten(long power)
+    {
+#if NET5_0
+      Debug.Assert(power < Constants.powers_of_ten_float.Length);
+      ref float tableRef = ref MemoryMarshal.GetArrayDataReference(Constants.powers_of_ten_float);
+      return Unsafe.Add(ref tableRef, (nint)power);
+#else
+      return Constants.powers_of_ten_float[power];
+#endif
+    }
+
+    public static unsafe float ToFloat(bool negative, AdjustedMantissa am)
     {
-      float d;
       ulong word = am.mantissa;
-      word |= (ulong)(am.power2) << FloatBinaryConstants.mantissa_explicit_bits;
+      word |= (ulong)(uint)(am.power2) << FloatBinaryConstants.mantissa_explicit_bits;
       word = negative ? word | ((ulong)(1) << FloatBinaryConstants.sign_index) : word;
 
-      unsafe
-      {
-        Buffer.MemoryCopy(&word, &d, sizeof(float), sizeof(float));
-      }
+      float d = 0;
+      Unsafe.Copy(ref d, &word);
```

Comment:
could we instead use ` float d = Unsafe.ReadUnaligned<float>(&word);`
It generates less code.

Reply:
Note that I have been using Unsafe.ReadUnaligned<float> in the ASCII PR.

Reply:
AFAIK it generates the same machine code. C# code is less, and that would be the obvious method to use (the whole copy-thing made me forgot to switch to this, trees and forests...).